tempesta_fw/t/functional/leaks/test_leak.py

```diff
@@ -0,0 +1,144 @@
+"""
+Testing for memory leaks
+"""
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+import unittest
+import re
+from time import sleep
+
+from testers import stress
+from helpers import tf_cfg, control, tempesta, remote
+
+def drop_caches():
+    """ Drop caches """
+    remote.tempesta.run_cmd("echo 3 > /proc/sys/vm/drop_caches")
+    sleep(1)
+
+def file_exists(remote_file):
+    """ Check existance of file on Tempesta host """
+    check_cmd = "if [ -e %s ]; then echo -n yes; fi" % remote_file
+    [stdout, stderr] = remote.tempesta.run_cmd(check_cmd)
+    if stdout != "yes":
+        return False
+    return True
+
+def has_kmemleak():
+    """ Check presence of kmemleak """
+    return file_exists("/sys/kernel/debug/kmemleak")
+
+def has_meminfo():
+    """ Check presence of meminfo """
+    return file_exists("/proc/meminfo")
+
+def read_kmemleaks():
+    """ Get amount of kmemleak records """
+    kmemleakfile = "/sys/kernel/debug/kmemleak"
+    if not has_kmemleak():
+        tf_cfg.dbg(1, "kmemleak file does not exists")
+        return -1
+    cmd = "cat %s | grep \"unreferenced object\" | wc -l" % kmemleakfile
+    [stdout, stderr] = remote.tempesta.run_cmd(cmd)
+    return int(stdout)
+
+def get_memory_line(name):
+    """ Get value from /proc/meminfo """
+    if not has_meminfo():
+        return -1
+    [stdout, stderr] = remote.tempesta.run_cmd("cat /proc/meminfo")
+    line = re.search("%s:[ ]+([0-9]+)" % name, stdout)
+    if line:
+        return int(line.group(1))
+    return -1
+
+def slab_memory():
+    """ Get amount of slab used memory """
+    drop_caches()
+    slabmem = get_memory_line("Slab")
+    return slabmem
+
+def used_memory():
+    """ Measure total memory usage """
+    drop_caches()
+    totalmem = get_memory_line("MemTotal")
+    if totalmem == -1:
+        return -1
+    freemem = get_memory_line("MemFree")
+    if freemem == -1:
+        return -1
+    return totalmem - freemem
+
+class LeakTest(stress.StressTest):
+    """ Leaks testing """
+    config = 'cache 0;\n'
+    backend_connections = 10
```

Comment:
Variable name doesn't correctly represent it's sense. `keepalive_requests` or something like that is more suitable naming.